vllm/v1/structured_output/backend_types.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: Apache-2.0
+
+import enum
+from abc import ABC, abstractmethod
+
+
+class StructuredOutputOptions(enum.Enum):
```

Comment:
a bit nit, but can we move this to `backends/`, then it can be `backends/abstract.py`, given that `types` is an actual library in python :)

Reply:
but then we'd have `backends/xgrammar.py`, clashing with `xgrammar` itself.  shrug

Reply:
😮‍💨 makes sense